shared-libs/transitions/src/transitions/index.js

```diff
@@ -285,10 +246,15 @@ const finalize = ({ change, results }, callback) => {
  * change/write.
  */
 const applyTransition = ({ key, change, transition }, callback) => {
+  if (!canRun({ key, change, transition })) {
+    logger.debug(
+      `canRun test failed on transition ${transition.key} for doc ${change.id} seq ${change.seq}`
+    );
+    return callback();
+  }
+
```

Comment:
I moved the call to `canRun` from the `applyTransitions` map function.
The reason was to test `canRun` over the document after it has been updated by previous transitions.

As an example, it enables the possibility of running `registration` over a fresh doc: currently `registration` requires either the form to be public or `doc.contact` to exist, while `update_clinics` is responsible for creating `doc.contact`.

Reply:
This is a good idea.

We should also formalise transition run order to make this less fragile.

Reply:
We already formalize that, since 2.15
https://github.com/medic/medic/blame/d7ec779eac1806a26cd6f546970b9319648a3cc7/shared-libs/transitions/src/transitions/index.js#L13